### PR TITLE
[JENKINS-66105] Allow follow redirects to work after 303 from `buildWithParameters`

### PR DESCRIPTION
**Pull request by jglick**
https://github.com/jenkinsci/jenkins/pull/5615#issuecomment-2215364164 is a source of confusion. Users not familiar with the details of how Jenkins queue item requests are supposed to be processed may assume that they can instruct an API client to follow redirects and that they will always get a 2xx status code when a build trigger request was somehow successful. After #5615, a fresh queue item would return a 201 with a `Location` header that a client would normally ignore; but an existing queue item would return a 303 with a `Location` that points to a URL not usable on its own, and a typical client will follow the redirect as is, yielding a 404 which left the mistaken impression that the queue item did not exist. Now we return a 200 to clarify that the queue item is there, though to be _useful_ you need to actually append to the returned URL to define a particular query you want (such as looking up the build in this example, which in turn would require a polling loop).

### Testing done

Added a test case that was missing from #5615. Without fix, fails in the reported confusing way (even though nothing is actually wrong):

```
org.htmlunit.FailingHttpStatusCodeException: 404 Not Found for http://localhost:43021/jenkins/queue/item/1/
	at org.htmlunit.WebClient.throwFailingHttpStatusCodeExceptionIfNecessary(WebClient.java:719)
	at org.htmlunit.WebClient.getPage(WebClient.java:473)
	at org.htmlunit.WebClient.getPage(WebClient.java:370)
	at org.htmlunit.WebClient.getPage(WebClient.java:523)
	at hudson.model.ParametersDefinitionPropertyTest.codeFromBuildWithParameters(ParametersDefinitionPropertyTest.java:118)
	at hudson.model.ParametersDefinitionPropertyTest.statusCodes(ParametersDefinitionPropertyTest.java:114)
```

### Proposed changelog entries

- `/queue/item/nnnnn/` URLs (with no `api/…` suffix) now return a 200 `text/plain` response to emphasize that the queue item does exist but that you would need to construct a specific query to find out more.

### Proposed upgrade guidelines

N/A

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/java/hudson/model/BuildAuthorizationToken.java
Comment (jglick) [inline core/src/main/java/hudson/model/BuildAuthorizationToken.java:87]: This logic was added in 0271fdb8a21ea6747a8b9a381274b45e13ca944b so that you could use `GET` requests with an API token to trigger a build. A year later, b2a98f6bc6924d1fd25f7da583888c2f4f36d83c broke that logic by removing the request property from `jenkins.security.ApiTokenProperty`. (Later, 814d202716a6c61c7d371c6a62755d296fe199a5 added a new property `jenkins.security.BasicHeaderApiTokenAuthenticator` for a similar purpose.)

Since no one seems to be complaining about the loss of this functionality, and we generally prefer to enforce `POST` on requests with side effects (`GET` requests with per-project tokens were grandfathered in), I am simply deleting the dead code to avoid confusion. My initial attempt at a test case in this PR failed because I thought that just passing an API token would be enough, but in fact `GET` does not work. (And there is no Java API to set the per-project token; you would have to configure it via an HTML form.)

File: core/src/main/java/jenkins/model/ParameterizedJobMixIn.java
Comment (jglick) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java:221]: Forgotten in #5615 or just “out of scope”? If such a change is desired, I would leave it to a separate PR for clarity.

**Review by Vlatombe — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]
Thanks!

This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

/label ready-for-merge

**Pull request merged**
